src/t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.cxx

```diff
@@ -0,0 +1,526 @@
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+#include <array>
+#include <sstream>
+
+#include <t8_geometry/t8_geometry_base.hxx>
+#include <t8_cmesh.hxx>
+#include <t8_cmesh/t8_cmesh_types.h>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.hxx>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.h>
+#include <t8_eclass.h>
+#include <t8_schemes/t8_default/t8_default_cxx.hxx>
+#include <t8_forest/t8_forest_general.h>
+#include <t8_forest/t8_forest_io.h>
+
+t8_geometry_lagrange::t8_geometry_lagrange (int dim)
+  : t8_geometry_with_vertices (dim, "t8_geom_lagrange_" + std::to_string (dim))
+{
+}
+
+t8_geometry_lagrange::~t8_geometry_lagrange ()
+{
+}
+
+void
+t8_geometry_lagrange::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreeid, const double *ref_coords,
+                                        const size_t num_points, double *out_coords) const
+{
+  if (num_points != 1)
+    SC_ABORT ("Error: Batch computation of geometry not yet supported.");
+  t8_geometry_lagrange::t8_geom_map (ref_coords, out_coords);
+}
+
+void
+t8_geometry_lagrange::t8_geom_evaluate_jacobian (t8_cmesh_t cmesh, t8_gloidx_t gtreeid, const double *ref_coords,
+                                                 const size_t num_points, double *jacobian) const
+{
+  SC_ABORT_NOT_REACHED ();
+}
+
+inline void
+t8_geometry_lagrange::t8_geom_load_tree_data (t8_cmesh_t cmesh, t8_gloidx_t gtreeid)
+{
+  t8_geometry_with_vertices::t8_geom_load_tree_data (cmesh, gtreeid);
+  degree = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (), T8_CMESH_LAGRANGE_POLY_DEGREE, gtreeid);
+  T8_ASSERT (degree != NULL);
+}
+
+void
+t8_geometry_lagrange::t8_geom_map (const double *ref_point, double *mapped_point) const
+{
+  const auto basis_functions = t8_geometry_lagrange::t8_geom_compute_basis (ref_point);
+  const size_t n_vertex = basis_functions.size ();
+  for (size_t i_component = 0; i_component < T8_ECLASS_MAX_DIM; i_component++) {
+    double inner_product = 0;
+    for (size_t j_vertex = 0; j_vertex < n_vertex; j_vertex++) {
+      const double coordinate = active_tree_vertices[j_vertex * T8_ECLASS_MAX_DIM + i_component];
+      const double basis_function = basis_functions[j_vertex];
+      inner_product += basis_function * coordinate;
+    }
+    mapped_point[i_component] = inner_product;
+  }
+}
```

Comment:
Because I thought `t8_geom_evaluate` would become more complicated, so I factored out the mapping into another function. However, `t8_geom_evaluate` is almost empty, so I will put the contents of `t8_geom_map` there.

Reply:
Then we should move the contents of `t8_geom_map` to `t8_geom_evaluate`